### PR TITLE
Add support to enable multicast on transit gateway

### PR DESCRIPTION
**Pull request by cschimid**
##### SUMMARY

Need to enable multicast while creating transit gateway

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

transit_gateway

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by cschimid** [comment]
I have successfully tested the module by creating a transit gateway with multicast enabled

File: CHANGELOG.rst [hunk 1]
```diff
@@ -4,6 +4,18 @@ community.aws Release Notes
 
 .. contents:: Topics
 
+v7.2.0
+======
```
Comment (alinabuzachis) [inline CHANGELOG.rst:8]: This isn't needed. The content of this file is automatically generated when a release is prepared.

File: plugins/modules/ec2_transit_gateway.py [hunk 2]
```diff
@@ -42,6 +42,11 @@
       - Whether to enable AWS DNS support.
     default: true
     type: bool
+  multicast_support:
+    description:
+      - Whether to enable AWS Multicast support.
+    default: true
+    type: bool
```
Comment (alinabuzachis) [inline plugins/modules/ec2_transit_gateway.py:49]: Please add `version_added: 7.3.0`

File: plugins/modules/ec2_transit_gateway_info.py [hunk 3]
```diff
@@ -119,6 +119,12 @@
                     returned: always
                     type: str
                     sample: "enable"
+                 multicast_support:
+                    description:
+                      - Indicates whether Multicast support is enabled.
+                    returned: always
+                    type: str
+                    sample: "enable"
```
Comment (alinabuzachis) [inline plugins/modules/ec2_transit_gateway_info.py:127]: Please add `version_added: 7.3.0`

**Review by alinabuzachis — COMMENTED** [review]
@cschimid Thank you for taking time to work on this feature. Can you please add a changelog fragment https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment and an integration test for this new feature? Thanks.

File: plugins/modules/ec2_transit_gateway.py [hunk 4]
```diff
@@ -181,6 +188,12 @@
           returned: always
           type: str
           sample: enable
+        multicast_support:
+          description: Indicates whether Multicast support is enabled.
+          returned: always
+          type: str
```
Comment (abikouo) [inline plugins/modules/ec2_transit_gateway.py:194]: Isn't it supposed to be a `bool` ?

File: plugins/modules/ec2_transit_gateway.py [hunk 5]
```diff
@@ -42,6 +42,12 @@
       - Whether to enable AWS DNS support.
     default: true
     type: bool
+  multicast_support:
+    description:
+      - Whether to enable AWS Multicast support. Valid only at the time of creation of the Transit Gateway.
+    default: false
```
Comment (alinabuzachis) [inline plugins/modules/ec2_transit_gateway.py:48]:
```suggestion

```

File: tests/integration/targets/ec2_transit_gateway/tasks/main.yml [hunk 6]
```diff
@@ -123,6 +123,7 @@
         filters:
           options.dns-support: enable
           options.vpn-ecmp-support: enable
+          options.multicast-support: enable
```
Comment (alinabuzachis) [inline tests/integration/targets/ec2_transit_gateway/tasks/main.yml:126]: I'm suggesting to remove `options.multicast-support: enable` from here and add:
- task to create a new ec2 transit gateway with multicast support (enabled or disabled)
- use the ec2_transit_gateway_info module with this filter options.multicast-support: enable or disable
- ensure the transit gateway you created is deleted inside the always block

File: plugins/modules/ec2_transit_gateway.py (hunk 4 above)
Comment (cschimid) [reply plugins/modules/ec2_transit_gateway.py:193]: > Isn't it supposed to be a `bool` ?

To configure is a bool, but return value is a string, sample "enable". https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_TransitGatewayRequestOptions.html

File: tests/integration/targets/ec2_transit_gateway/tasks/main.yml (hunk 6 above)
Comment (cschimid) [reply tests/integration/targets/ec2_transit_gateway/tasks/main.yml:126]: Done, I inserted these two parts:

    - name: test create transit gateway with multicast enabled
      ec2_transit_gateway:
        description: "{{ tgw_description }}"
        multicast_support: true
      register: create_result
    - name: assert changed is True
      assert:
        that:
          - create_result.changed == True
          
    - name: test success with multicast enabled
      ec2_transit_gateway_info:
        filters:
          options.multicast-support: enable
      register: result
    - name: assert success with transit_gateway_id filter
      assert:
        that:
          - 'result.changed == false'
          - 'result.transit_gateways != []

File: tests/integration/targets/ec2_transit_gateway/tasks/main.yml [hunk 7]
```diff
@@ -129,6 +140,7 @@
         that:
           - 'result.changed == false'
           - 'result.transit_gateways != []'
+
```
Comment (alinabuzachis) [inline tests/integration/targets/ec2_transit_gateway/tasks/main.yml:143]: You could then modify this part this way:
```
 always:
    ###### TEARDOWN STARTS HERE ######
    - name: delete transit gateway 
      ec2_transit_gateway:
        description: "{{ item }}"
        state: absent
      ignore_errors: yes
      loop:
       - "{{ tgw_description }}"
       - "{{ tgw_description_multicast }}"

```

File: CHANGELOG.rst [hunk 8]
```diff
@@ -4,29 +4,6 @@ community.aws Release Notes
 
 .. contents:: Topics
 
-v7.2.0
```
Comment (alinabuzachis) [inline CHANGELOG.rst:7]: @cschimid You should restore part.

**Review by alinabuzachis — APPROVED** [review]

File: plugins/modules/ec2_transit_gateway.py [hunk 9]
```diff
@@ -42,6 +42,11 @@
       - Whether to enable AWS DNS support.
     default: true
     type: bool
+  multicast_support:
+    description:
+      - Whether to enable AWS Multicast support. Valid only at the time of creation of the Transit Gateway.
+    type: bool
+    version_added: 7.3.0
```
Comment (tremble) [inline plugins/modules/ec2_transit_gateway.py:49]:
```suggestion
    version_added: 8.1.0
```

**Review by abikouo — APPROVED** [review]

**Review by tremble — APPROVED** [review]

**Pull request merged**